package/kedro_viz/server.py

```diff
@@ -474,7 +489,39 @@ def format_pipeline_data(
 @app.route("/api/nodes.json")
 def nodes_json():
     """Serve the pipeline data."""
-    return jsonify(data)
+    return jsonify(_DATA)
+
+
+@app.route("/api/nodes/<string:node_id>")
+def nodes_metadata(node_id):
+    """Serve the metadata for node and dataset."""
+    node = _NODES.get(node_id)
+    if node and node.get("type") == "task":
+        task_metadata = {
+            # pylint: disable=protected-access
+            "code": inspect.getsource(node["obj"]._func)
+        }
+        # pylint: disable=protected-access
+        code_full_path = Path(inspect.getfile(node["obj"]._func)).expanduser().resolve()
+        code_location = Path(code_full_path).relative_to(Path.cwd().parent)
+        task_metadata.update({"code_location": str(code_location)})
+
+        # pylint: disable=protected-access
+        docstring = inspect.getdoc(node["obj"]._func)
+        if docstring:
+            task_metadata.update({"docstring": docstring})
+        return jsonify(task_metadata)
+    if node and node.get("type") == "data":
+        dataset = node["obj"]
+        dataset_metadata = {
+            "dataset_type": dataset.__module__,
```

Comment:
I think you need `dataset.__class__.__module__` and `dataset.__class__.__qualname__`

Reply:
Thanks! I tried different combinations of class modules, and got this result (very interesting!!)

![Screenshot 2020-08-06 at 14 57 46](https://user-images.githubusercontent.com/8097799/89540624-37b83580-d7f5-11ea-90af-07719bffa35d.png)


So seems like we need `dataset.__module__` and `dataset.__qualname__`

Reply:
Never mind, you are right. The above examples are from class, not an instance of the class. 


![Screenshot 2020-08-06 at 15 53 05](https://user-images.githubusercontent.com/8097799/89546872-f6c41f00-d7fc-11ea-9cfb-27a9cd0a8a15.png)